crypto/info.c

```diff
@@ -10,8 +10,56 @@
 #include <stddef.h>
 #include <openssl/crypto.h>
 #include "internal/dso_conf.h"
+#include "internal/thread_once.h"
 #include "e_os.h"
 
+#if defined(__i386)   || defined(__i386__)   || defined(_M_IX86) || \
+    defined(__x86_64) || defined(__x86_64__) || \
+    defined(_M_AMD64) || defined(_M_X64)
+# include "internal/cryptlib.h"
+#elif defined(__arm__) || defined(__arm) || defined(__aarch64__)
+# include "arm_arch.h"
+#endif
+
+static const char* ossl_cpu_info_str = NULL;
+static CRYPTO_ONCE init_cpu_info = CRYPTO_ONCE_STATIC_INIT;
+
+DEFINE_RUN_ONCE_STATIC(ossl_init_cpu_info_str)
```

Comment:
I think it's a mistake to name the RUN_ONCE stuff specifically for CPU info, as we might have more strings to fill in...  see #9689

Reply:
so i can call this do_init_info
and the once thinf, ossl_init_info_once ?

Reply:
Well, if you look in #9689, you can see that I propose `init_info` and `init_info_strings`, respectively (since the function is static, I don't see any reason to prefix it with `ossl_`)

Reply:
okay.